Sources/CryptomatorCloudAccess/WebDAV/WebDAVSession.swift

```diff
@@ -233,14 +233,15 @@ class WebDAVSession {
 		}
 	}
 
-	func performUploadTask(with request: URLRequest, fromFile fileURL: URL) -> Promise<(HTTPURLResponse, Data?)> {
+	func performUploadTask(with request: URLRequest, fromFile fileURL: URL, onTaskCreation: ((URLSessionUploadTask?) -> Void)?) -> Promise<(HTTPURLResponse, Data?)> {
 		HTTPDebugLogger.logRequest(request)
 		let progress = Progress(totalUnitCount: 1)
 		let task = urlSession.uploadTask(with: request, fromFile: fileURL)
 		progress.addChild(task.progress, withPendingUnitCount: 1)
 		let pendingPromise = Promise<(HTTPURLResponse, Data?)>.pending()
 		let webDAVDataTask = WebDAVDataTask(promise: pendingPromise)
 		delegate?.addRunningDataTask(key: task, value: webDAVDataTask)
+		onTaskCreation?(task)
```

Comment:
Is there a reason why it's not called right after the task has been created (in line 239)? That's how you've done it for the download task.

Reply:
No specific reason :D I'll change it 👍

Reply:
Resolved by https://github.com/cryptomator/cloud-access-swift/pull/22/commits/2a3662c0ca6341215fd5c1e9c045e7fd2352dd14.